### PR TITLE
Workloads clusters docs

### PR DESCRIPTION
**Pull request by dekiel**
Documentaition about shiny new workload clusters.

**Related issue(s)**
See #2482

### CODE REVIEW

File: docs/prow/workload-clusters.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+```gcloud container clusters list
+   NAME                          LOCATION        MASTER_VERSION  MASTER_IP       MACHINE_TYPE   NODE_VERSION    NUM_NODES  STATUS
+   trusted-workload-kyma-prow    europe-west3    1.14.10-gke.36  _____________   n1-standard-4  1.14.10-gke.36  3          RUNNING
+   untrusted-workload-kyma-prow  europe-west3    1.14.10-gke.36  _____________   n1-standard-4  1.14.10-gke.36  2          RUNNING
+```
+
+## Infrastructure design
+
+Cluster are located in separate networks for trusted and untrusted components. Each networks provide three subnets for cluster nodes, pods and services.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:22]:
```suggestion
Clusters are located in separate networks for trusted and untrusted components. Each network provides three subnets for cluster nodes, Pods, and services.
```

File: docs/prow/workload-clusters.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+There is no peering between networks, thus clusters are isolated on network level.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:23]:
```suggestion
There is no peering between networks, thus clusters are isolated on the network level.
```

File: docs/prow/workload-clusters.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+Each cluster has dedicated Cloud Router with CloudNAT and external IP. This provide outgoing connectivity for clusters and fixed external IP from which all traffic is seen.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:24]:
```suggestion
Each cluster has a dedicated Cloud Router with Cloud NAT and an external IP address. This provides outgoing connectivity for clusters and a fixed external IP address that receives all traffic.
```

File: docs/prow/workload-clusters.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+untrusted-workload-kyma-prow  europe-west3  untrusted-kyma-prow
+
+gcloud compute addresses list
+NAME                          ADDRESS/RANGE   TYPE      PURPOSE  NETWORK  REGION        SUBNET  STATUS
+trusted-workload-kyma-prow    _____________   EXTERNAL                    europe-west3          IN_USE
+untrusted-workload-kyma-prow  _____________   EXTERNAL                    europe-west3          IN_USE
+```
+## Prow design
+
+Prow is accessing workload clusters using x509 client certificates with cluster admin role.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:49]:
```suggestion
Prow accesses workload clusters using X.509 client certificates and the **cluster-admin** role.
```

Comment (kazydek) [reply docs/prow/workload-clusters.md:49]: (or simply pls specify which role you mean exactly and correct me if I'm wrong with my assumption :) )

File: docs/prow/workload-clusters.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Certificates are combined in to kubeconfig file and stored as a secret on prow cluster.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:50]:
```suggestion
Certificates are combined into a kubeconfig file and stored as a secret on a Prow cluster.
```

File: docs/prow/workload-clusters.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+Jobs use context names to indicate target workload cluster to run.
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:51]:
```suggestion
Jobs use context names to indicate the target workload cluster to run.
```

File: docs/prow/workload-clusters.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+```
+k config get-contexts --kubeconfig workload-clusters-kubeconfig.yaml
+CURRENT   NAME                 CLUSTER              AUTHINFO             NAMESPACE
+*         default              default              default
+          trusted-workload     trusted-workload     trusted-workload
+          untrusted-workload   untrusted-workload   untrusted-workload
+```
+For details about building kubeconfig file and providing it to prow see [upstream documentation](https://github.com/kubernetes/test-infra/tree/master/gencred).
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:60]:
```suggestion
For details about building the kubeconfig file and sending it to Prow, see the official [k8s documentation](https://github.com/kubernetes/test-infra/tree/master/gencred).
```

Comment (dekiel) [reply docs/prow/workload-clusters.md]: I would keep word "providing" instead "sending". After storing kubeconfig on a cluster, you need to let Prow know where it is.

File: docs/prow/workload-clusters.md (hunk 3 above)
Comment (dekiel) [reply docs/prow/workload-clusters.md:24]: This IP is not receiving all traffic. It's used as a source IP for all traffic initiated from pods or worker nodes. What about, ".. IP address from which, all traffic originated from pods and worker nodes is seen."?

File: docs/prow/workload-clusters.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
+# Prow workload clusters
+
+It's a description of prow workload clusters. These are clusters where prow is scheduling pods which execute prowjob logic. All workload clusters are aggregated under kyma-prow GCP project. We use two workload clusters for trusted and untrusted prowjobs.
+
+## Clusters design
+
+Clusters have enabled autoupgrade and follow stable channel.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply docs/prow/workload-clusters.md]: Applied in recent commit.

File: docs/prow/workload-clusters.md (hunk 3 above)
Comment (kazydek) [reply docs/prow/workload-clusters.md:25]: Ok, I just changed passive voice into active voice - see my updated suggestion above :)

File: docs/prow/workload-clusters.md (hunk 7 above)
Comment (kazydek) [reply docs/prow/workload-clusters.md]: Ok

Comment (dekiel) [reply docs/prow/workload-clusters.md]: Applied in commit 90893ea.

File: docs/prow/workload-clusters.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+Jobs use context names to indicate the target workload cluster to run.
+
+```
+k config get-contexts --kubeconfig workload-clusters-kubeconfig.yaml
+CURRENT   NAME                 CLUSTER              AUTHINFO             NAMESPACE
+*         default              default              default
+          trusted-workload     trusted-workload     trusted-workload
+          untrusted-workload   untrusted-workload   untrusted-workload
+```
+For details about building the kubeconfig file and providing it to Prow see the official [k8s documentation](https://github.com/kubernetes/test-infra/tree/master/gencred).
```
Comment (kazydek) [inline docs/prow/workload-clusters.md:61]:
```suggestion
For details about building the kubeconfig file and providing it to Prow, see the official [k8s documentation](https://github.com/kubernetes/test-infra/tree/master/gencred).
```

**Pull request merged**